src/test/kotlin/com/exactpro/th2/codec/DecodeProcessorTest.kt

```diff
@@ -18,99 +18,84 @@ package com.exactpro.th2.codec
-        val batch = MessageGroupBatch.newBuilder().apply {
-            addGroups(MessageGroup.newBuilder().apply {
-                this += RawMessage.newBuilder().apply {
-                    metadataBuilder.apply {
-                        id = MESSAGE_ID
-                        protocol = originalProtocol
-                    }
-                }
-                this += RawMessage.newBuilder().apply {
-                    metadataBuilder.apply {
-                        id = MESSAGE_ID
-                        protocol = wrongProtocol
-                    }
-                }
-                this += Message.newBuilder().apply {
-                    messageType = "test-type"
-                    metadataBuilder.apply {
-                        id = MESSAGE_ID
-                        protocol = originalProtocol
-                    }
-                }
-                this += RawMessage.newBuilder().apply {
-                    metadataBuilder.apply {
-                        id = MESSAGE_ID
-                        protocol = originalProtocol
-                    }
-                }
-                this += RawMessage.getDefaultInstance()
-            }.build())
-        }.build()
-
+        val processor = UniversalCodecProcessor(TestCodec(true), originalProtocols, CODEC_EVENT_ID.toProto(), false, process = DECODE, protocol = protocol) {}
+        val batch = getNewBatchBuilder(protocol, BOOK_NAME, SESSION_GROUP_NAME)
+            .addNewRawMessage(MESSAGE_ID, protocol = originalProtocol)
+            .addNewRawMessage(MESSAGE_ID, protocol = wrongProtocol)
+            .addNewParsedMessage(MESSAGE_ID, type = MESSAGE_TYPE, protocol = originalProtocol)
+            .addNewRawMessage(MESSAGE_ID, protocol = originalProtocol)
+            .addNewRawMessage()
+            .build()
 
         val result = processor.process(batch)
 
-        Assertions.assertEquals(5, result.getGroups(0).messagesList.size) {"group of outgoing messages must be the same size"}
+        Assertions.assertEquals(5, result.groupCursor.messagesCount) { "group of outgoing messages must be the same size" }
 
-        Assertions.assertTrue(result.getGroups(0).messagesList[0].hasMessage())
-        result.getGroups(0).messagesList[0].message.let {
-            Assertions.assertEquals(ERROR_TYPE_MESSAGE, it.messageType)
-            Assertions.assertEquals(originalProtocol, it.metadata.protocol)
-        }
+        result.groupCursor.messageCursor.let {
+            assertTrue(it.isParsed)
+            Assertions.assertEquals(ERROR_TYPE_MESSAGE, it.asParsed.type)
+            Assertions.assertEquals(originalProtocol, it.protocol)
 
-        Assertions.assertTrue(result.getGroups(0).messagesList[1].hasRawMessage())
-        result.getGroups(0).messagesList[1].rawMessage.let {
-            Assertions.assertEquals(wrongProtocol, it.metadata.protocol)
-        }
+            result.groupCursor.currentMessageIndex = 1
```

Comment:
Probably, we should correct the cursor API because those calls with indexes look very odd... I would think that the cursor should have some kind of `next` and `prev` methods to move back and forward through the messages